int64 fruit_cnt
